networkit/cpp/scd/CliqueDetect.cpp

```diff
@@ -0,0 +1,158 @@
+#include <unordered_map>
+
+#include <networkit/auxiliary/UniformRandomSelector.hpp>
+#include <networkit/clique/MaximalCliques.hpp>
+#include <networkit/graph/GraphBuilder.hpp>
+#include <networkit/scd/CliqueDetect.hpp>
+
+namespace NetworKit {
+
+CliqueDetect::CliqueDetect(const Graph &G) : SelectiveCommunityDetector(G) {
+    if (G.numberOfSelfLoops() > 0)
+        throw std::runtime_error("CliqueDetect works only with simple graphs.");
+}
+
+std::set<node> CliqueDetect::expandOneCommunity(node s) {
+    // 1. get the maximum clique in neighbors(s) + s
+    std::set<node> result;
+    result.insert(s);
+
+    std::vector<node> sn;
+    sn.reserve(G->degree(s));
+    G->forNeighborsOf(s, [&](node v) { sn.push_back(v); });
```

Comment:
What about `std::vector<node> sn(G->neighborRange(s).begin(), G->neighborRange(s).end())`

Reply:
The code predates the introduction of `neighborRange`, I agree that it is much nicer and also changed the remainder of this algorithm to use neighbor ranges. I also discovered a bug in the implementation for weighted graphs that I fixed.